### ISSUE TITLE
Error with repo.packagist.org/packages.json

### ISSUE THREAD

**Issue by nlhommet** [composer/composer#7493]
I have the following error with composer 1.7-RC when I do composer update
```
 [Seld\JsonLint\ParsingException]
  "https://repo.packagist.org/packages.json" does not contain valid JSON
  Parse error on line 1:
  ▼\\\\\\♥��ݎ♀���
  ^
  Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
```

**Comment by Seldaek** [comment]
Do you consistently see this? Do you not get it with 1.6.5? Can you run `composer diagnose` and paste the full output here as well as full output of `composer update -vvv`?

**Comment by nlhommet** [comment]
With 1.6.5 `composer update` work fine.
With composer 1.7-RC :
`composer diagnose` : 
```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.7.0-RC
PHP version: 7.1.8
PHP binary path: D:\library\php7.1.8\php.exe
```
`composer update -vvv` : 
```
Reading ./composer.json
Loading config file C:/Users/crazy/AppData/Roaming/Composer/auth.json
Loading config file ./composer.json
Checked CA file C:\Users\crazy\AppData\Local\Temp\opeFD8D.tmp: valid
Executing command (D:\projets\gipcaen\api): git branch --no-color --no-abbrev -v
Reading C:/Users/crazy/AppData/Roaming/Composer/composer.json
Loading config file C:/Users/crazy/AppData/Roaming/Composer/auth.json
Loading config file C:/Users/crazy/AppData/Roaming/Composer/composer.json
Loading config file C:\Users\crazy\AppData\Roaming\Composer/auth.json
Reading C:\Users\crazy\AppData\Roaming\Composer/auth.json
Reading D:\projets\gipcaen\api/vendor/composer/installed.json
Reading C:/Users/crazy/AppData/Roaming/Composer/vendor/composer/installed.json
Loading plugin PackageVersions\Installer
Loading plugin Symfony\Flex\Flex
Downloading https://repo.packagist.org/packages.json
Downloading https://repo.packagist.org/packages.json
Downloading https://repo.packagist.org/packages.json


  [Seld\JsonLint\ParsingException]
  "https://repo.packagist.org/packages.json" does not contain valid JSON
  Parse error on line 1:
  ▼\\\\\\♥���n←G♀��E
  ^
  Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['


Exception trace:
 () at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Json/JsonFile.php:299
 Composer\Json\JsonFile::validateSyntax() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Json/JsonFile.php:272
 Composer\Json\JsonFile::parseJson() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Repository/ComposerRepository.php:701
 Composer\Repository\ComposerRepository->fetchFile() at D:\projets\gipcaen\api\vendor\symfony\flex\src\ComposerRepository.php:56
 Symfony\Flex\ComposerRepository->fetchFile() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Repository/ComposerRepository.php:497
 Composer\Repository\ComposerRepository->loadRootServerFile() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Repository/ComposerRepository.php:245
 Composer\Repository\ComposerRepository->getProviderNames() at D:\projets\gipcaen\api\vendor\symfony\flex\src\Flex.php:673
 Symfony\Flex\Flex->Symfony\Flex\{closure}() at D:\projets\gipcaen\api\vendor\symfony\flex\src\ParallelDownloader.php:246
 Symfony\Flex\ParallelDownloader->getNext() at D:\projets\gipcaen\api\vendor\symfony\flex\src\ParallelDownloader.php:84
 Symfony\Flex\ParallelDownloader->download() at D:\projets\gipcaen\api\vendor\symfony\flex\src\Flex.php:674
 Symfony\Flex\Flex->populateRepoCacheDir() at D:\projets\gipcaen\api\vendor\symfony\flex\src\Flex.php:208
 Symfony\Flex\Flex->activate() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginManager.php:236
 Composer\Plugin\PluginManager->addPlugin() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginManager.php:205
 Composer\Plugin\PluginManager->registerPackage() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginManager.php:261
 Composer\Plugin\PluginManager->loadRepository() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Plugin/PluginManager.php:76
 Composer\Plugin\PluginManager->loadInstalledPlugins() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Factory.php:384
 Composer\Factory->createComposer() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Factory.php:576
 Composer\Factory::create() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:341
 Composer\Console\Application->getComposer() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:454
 Composer\Console\Application->getPluginCommands() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:152
 Composer\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:103
 Composer\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:56
 require() at C:\ProgramData\ComposerSetup\bin\composer.phar:24
```

**Comment by Seldaek** [comment]
What if you do `composer update --no-plugins` is it also failing?

**Comment by nlhommet** [comment]
same thing

**Comment by Seldaek** [comment]
@nlhommet thanks. Any chance you could install composer from a git clone and try to debug this further? I can't reproduce here.. Would be good if you could check out the 1.6.5 tag locally and run a composer update in it to get the latest versions of our dependencies see if any dependency update is causing it or if it's in one of the changes we did in the composer code in 1.7

**Comment by nlhommet** [comment]
OK I try to debug this on my side

**Comment by Seldaek** [comment]
Also one thing worth trying is to `composer self-update --snapshot` and then try again `composer update --no-plugins` to make sure it's not flex. There was a bug in 1.7-RC which made --no-plugins not quite work all the time..

**Comment by nlhommet** [comment]
I have try to debug  file `src/Composer/Util/RemoteFilesystem.php`
And when i replace line 318
```
 list($http_response_header, $result) = $this->getRemoteContents($originUrl, $fileUrl, $ctx);
```
 by
```
if (!\preg_match('`https`', $originUrl)) {
        $originUrl = 'https://' . $originUrl;
}
$result = \file_get_contents($originUrl, false, $ctx);
```
It's working.
But sometimes I get a HTML file from `https://repo.packagist.org/packages.json`

**Comment by Seldaek** [comment]
You should use `$fileUrl` there not $originUrl. `$fileUrl` should contain https:// already. I don't really see what your change does otherwise..

Pinging @nicolas-grekas as he did this change.. One thing I notice is that the catch block there at line 357 doesn't have access to `$http_response_header` as the new getRemoteContents method will never return it in that scope.. This is problematic (but maybe not the cause of this issue).

**Comment by nlhommet** [comment]
Ok, I change line 318 by `$result = \file_get_contents($fileUrl, false, $ctx);` and everything works well

**Comment by nicolas-grekas** [comment]
@nlhommet can you please check #7495

**Comment by staabm** [comment]
We see the same error in travis builds
https://travis-ci.org/hirak/prestissimo/jobs/408381731

**Comment by nlhommet** [comment]
I check that tomorrow as soon as I get to work

**Comment by staabm** [comment]
I would guess the error is related to one/a few of the new repo.packagist.org server nodes, not the composer client

**Comment by nicolas-grekas** [comment]
you need to upgrade symfony/flex, see https://github.com/symfony/flex/issues/397

**Comment by nlhommet** [comment]
I went rollback to the version 1.6.5
After I executed : 
```
> composer update --no-plugins

Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Updating symfony/flex (v1.0.85 => v1.0.86): Downloading (100%)
Writing lock file
Generating autoload files

> composer self-update --snapshot
Updating to version 35389f1cc2ef3abf2a069a21902fb681302e0217 (snapshot channel).
   Downloading (100%)
Use composer self-update --rollback to return to version 1.6.5

> composer update
"https://repo.packagist.org/packages.json" does not contain valid JSON
Parse error on line 1:
▼\\\\\\♥��ݎ♀���
^
Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
https://repo.packagist.org could not be fully loaded, package information was loaded from the local cache and may be out of date

> composer clearcache
Clearing cache (cache-vcs-dir): C:\Users\crazy\AppData\Local\Composer\vcs
Clearing cache (cache-repo-dir): C:\Users\crazy\AppData\Local\Composer\repo
Clearing cache (cache-files-dir): C:\Users\crazy\AppData\Local\Composer\files
Clearing cache (cache-dir): C:\Users\crazy\AppData\Local\Composer
All caches cleared.

> composer update
 [Composer\Repository\RepositorySecurityException]
The contents of https://repo.packagist.org/p/provider-2013%24e00fadef42535a8348d8f5edccb4a7dd892ac3bf8477c073ec03eb16f1cef7d7.json do not match its signature. This could indicate a man-in-the-middle attack or e.g. antivirus software corrupting files. Try running composer again and report this if you think it is a mistake.

> composer update
"https://repo.packagist.org/packages.json" does not contain valid JSON
Parse error on line 1:
▼\\\\\\♥��ݎ♀���
^
Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
https://repo.packagist.org could not be fully loaded, package information was loaded from the local cache and may be out of date
````

**Comment by nlhommet** [comment]
same thing with `symfony/flex v1.0.87`

**Comment by nicolas-grekas** [comment]
@staabm the failure you linked looks unrelated: the version that runs on Travis is v1.6.5, which doesn't have the new "getRemoteContents()" method, isn't it?

**Comment by staabm** [comment]
For me it looks like the failure I linked is the same as the initially reported one. 

I agree it is most likely not related to the error you fixes in your PR

**Comment by nicolas-grekas** [comment]
@nlhommet I'm unable to reproduce the issue on my Windows VM. Would you be able to share a VM that reproduce the issue (can be shared privately using e.g. a http://send.firefox.com/ linked you'd give me on Slack?)
Alternatively, could you describe your environment precisely (e.g. output of a phpinfo()) + your locale `echo setlocale(LC_ALL, 0);`?

**Comment by nicolas-grekas** [comment]
Also, can you try actually *removing* flex? Does the issue still exist once it's been completely removed?

**Comment by nlhommet** [comment]
My phpinfo : [phpinfo.txt](https://github.com/composer/composer/files/2235793/phpinfo.txt)
My locale : `LC_COLLATE=C;LC_CTYPE=French_France.1252;LC_MONETARY=C;LC_NUMERIC=C;LC_TIME=C`
It's strange, `composer install` works perfectly but `composer update` not...
When I remove `symfony/flex` : 

```
>composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing symfony/flex (v1.0.87):
"https://repo.packagist.org/packages.json" does not contain valid JSON
Parse error on line 1:
▼\\\\\\♥���n←G♀��e
^
Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
https://repo.packagist.org could not be fully loaded, package information was loaded from the local cache and may be out of date
Generating autoload files
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Executing script cache:clear [OK]
Executing script assets:install public [OK]
```

**Comment by nicolas-grekas** [comment]
>   - Installing symfony/flex (v1.0.87):
can you remove symfony/flex from the composer.json entirely?

**Comment by nlhommet** [comment]
When I remove completely `symfony/flex` (vendors and composer.json), `composer update` works!!!

**Comment by nicolas-grekas** [comment]
Great, so I suppose we can close here, the issue is in flex...

**Comment by nlhommet** [comment]
It's strange, with `composer 1.6.5` and `symfony/flex` all working

**Comment by nlhommet** [comment]
that explains why when I move all the code in `getRemoteContents` at line 318 in `RemoteFilesystem` all working.

**Comment by nicolas-grekas** [comment]
I managed to reproduce the issue, see fix in https://github.com/symfony/flex/pull/401

**Comment by Seldaek** [comment]
Cool that you found a fix. Does that mean it writes to temp file instead of
keeping it in memory?

**Comment by nicolas-grekas** [comment]
It keeps the first 32K of headers and 512K of compressed in memory, then writes to disk yes.

**Comment by nlhommet** [comment]
I would test monday with version `1.0.88` of symfony/flex but I think the problem is solved with the PR of @nicolas-grekas

**Comment by staabm** [comment]
One note again: I can see similar errors on travis builds in non symfony/flex context

https://travis-ci.org/hirak/prestissimo/builds/409040647?utm_source=github_status&utm_medium=notification

**Comment by nicolas-grekas** [comment]
@staab your issue looks totally unrelated. In your case the compressed content is output instead of being buffered. This issue is about json payloads being corrupted. I'd be surprised if this ends up being a Composer issue. Please look closer on your side...

**Comment by nlhommet** [comment]
Solved with symfony/flex `1.0.89`

**Comment by Tanvirmehedi** [comment]
> I merged #7495 so you can try out the fix by doing `composer self-update --snapshot`

Thank You This command Is work For me
